api/views/event.py

```diff
@@ -28,3 +29,27 @@ def get_event_types(self, request, pk=None):
         Returns a dictionary of the defined event types
         """
         return Response(EVENT_TYPE_CHOICES)
+
+    @action(detail=False, methods=['post'], url_path="duplicate-event")
+    def duplicate_event(self, request):
+        """
+        Duplicates an event as well as the questions within the event.
+        """
+        event = Event.objects.filter(id=request.data.get("event", None)).first()
+        course = CanvasCourse.objects.filter(id=request.data.get("course", None)).first()
+        old_event_id = event.id
+        event.id = None
+        event.course = course
+        event.name += ' (Copy)'
+        event.save()
+        for question in Question.objects.all().filter(event=old_event_id):
+            question.id = None
+            question.pk = None
+            question.question_ptr_id = None
+            question.course = course
+            question.event = event
+            question.title += ' (Copy)'
+            question.save()
+        return Response({
+            "success": True,
```

Comment:
We are deprecating the `success` we should check the response status instead. Here i think you can either return the new event or nothing.